### PR TITLE
feat: implement contract migration to add projects to pagination

### LINKED ISSUE #331: List all projects
We could have a way to list all projects.

If we want to have a full on-chain way, we need to change the data structure or extend it. We can make a pagination system (similar to what we have for the DAO pages) to list keys and names. That would not be too much duplication of data.

Or we don't do this on-chain and use an indexer.

What I don't want to for now is to have our DB and backend. I want to keep a full decentralised mode of operation for as long as we can.

### PR DESCRIPTION
**Pull request by masch**
# 🚀 Migration of Existing Projects to Pagination System

## Description
This PR implements the `MigrationTrait` to allow administrators to manually link existing orphaned projects (projects that exist in persistent storage but are not yet tracked in the pagination lists) into the contract's pagination system.

## 🔗 Related Issue
Closes #331

### Core Logic
- **Name-based Migration**: The implementation in allows passing plain project names (`Vec<String>`).
- **Internal Hashing**: Automatically calculates project keys from names using Keccak256 to match the registration system.
- **Improved Duplication Check**: Replaced manual loops with idiomatic `.contains()` checks for project keys.

### Refactoring & Security
- **Shared Constants**: Moved `MAX_PROJECTS_PER_PAGE` to `types.rs` to unify pagination logic across `contract_versioning` and `contract_migration`.
- **Administrative Controls**: 
  - Integrated `auth_admin` check to restrict usage to authorized administrators.
  - Added `require_not_paused` check to prevent migrations during maintenance modes.

### Automated Tests
A new test suite [`test_migration.rs`](contracts/tansu/src/tests/test_migration.rs) was added to verify all edge cases:
- test_add_projects_to_pagination_previously_registered_project: Verifies migration of orphaned keys and duplicate prevention.
- test_add_projects_to_pagination_unregistered_project: Ensures non-existent project names are ignored.
- test_add_projects_to_pagination_paused_error: Validates respect for the contract's paused state.
- test_add_projects_to_pagination_unauthorized_admin_error: Confirms only admins can execute the migration.

### CODE REVIEW

File: contracts/tansu/src/contract_versioning.rs [hunk 1]
```diff
@@ -50,15 +47,7 @@ impl VersioningTrait for Tansu {
...
-        let str_len = name.len() as usize;
-        if str_len > 15 {
-            // could add more checks but handled in any case with later calls
-            panic_with_error!(&env, &errors::ContractErrors::InvalidDomainError);
-        }
-        let mut slice: [u8; 15] = [0; 15];
-        name.copy_into_slice(&mut slice[..str_len]);
-        let name_b = Bytes::from_slice(&env, &slice[0..str_len]);
-
+        let name_b = crate::get_name_bytes(&env, &name);
```
Comment (tupui) [inline contracts/tansu/src/contract_versioning.rs]: No don't do any refactoring for a migration. Migrations are one-off scripts and must not drive the business logic. Same goes for tests. If we need to re-do something, it's perfectly fine. The other issue is that doing so breaks code lineage.

**Review by tupui — CHANGES_REQUESTED** [review]
Thanks for working on this. We can drastically simplify the approach. I left some comments.

A quick note on AI. Avoid letting it write comments for PR descriptions, or even in general. It never works besides being overly confident that it did things perfectly and adding useless emoji all over the place 😅

File: contracts/tansu/src/lib.rs [hunk 2]
```diff
@@ -208,3 +213,40 @@ fn validate_contract(env: &Env, contract: &types::Contract) {
...
+/// # Arguments
+/// * `env` - The environment object
+/// * `name` - The name to get the project key from
+///
+/// # Returns
+/// * `Bytes` - The project key
+///
+/// # Panics
+/// * If the name is longer than 15 bytes
+pub fn get_project_key(env: &Env, name: &String) -> Bytes {
```
[the remark this answers is not in the document]
Comment (masch) [reply contracts/tansu/src/lib.rs]: I did an extra refactor earlier to avoid duplicating the function that generates project keys. I've reverted the commit.

File: contracts/tansu/src/contract_versioning.rs (hunk 1 above)
Comment (masch) [reply contracts/tansu/src/contract_versioning.rs]: Reverted, I was trying to avoid duplication.

**Comment by masch** [comment]
> Thanks for working on this. We can drastically simplify the approach. I left some comments.
> 
> A quick note on AI. Avoid letting it write comments for PR descriptions, or even in general. It never works besides being overly confident that it did things perfectly and adding useless emoji all over the place 😅

I manually added the comments in the test to explain the logic, and I followed the existing pattern for the trait function definitions. Regarding the PR comments, you’re right—those were AI-generated suggestions.

Let me know if this looks like the right approach for the project! 🎅

**Comment by masch** [comment]
I’ve added support for multiple pages in the migration. I also added a check to see if the project name already exists on any page; although that’s a very rare edge case, it’s a good safety measure to have.
